chainerx/_docs/routines.py

```diff
@@ -807,6 +807,13 @@ def _docs_linalg():
 Returns:
     :class:`~chainerx.ndarray`: Output array. Cholesky factor of ``a``.
 
+Note:
+    This functions assumes that ``a`` is symmetric and is undefined otherwise.
+
+Note:
+    The computed gradient is a symmetric matrix and not an upper/lower
+    triangular matrix.
```

Comment:
I suggest:

```The computed gradient is a symmetric matrix, where the mathematical definition of Chelesky decomposition supersedes the definition of ``*potrf`` in LAPACK. More specifically, the gradient is computed as if the domain of the function is restricted to a Riemannian submanifold of the Euclidean space :math:`R^{n \times n}` (an open subset of the subspace of symmetric matrices, whose inner product structure is compatible with the Frobenius norm).```

(`cholesky` needs "an open subset of" for the positive-definiteness but `eigh` does not.)

Reply:
@masomatics helped me to improve the above.

The computed gradient is a symmetric matrix. Our implementation of backpropagation is faithful to the mathematical definition of Cholesky decomposition; in forward computation with LAPACK, which does not really check if the input matrix is symmetric--- we, however, do not really backpropagate this very process.  Instead, the gradient is computed as if the function is restricted to a Riemannian submanifold of R_{n \times n} consisting just of positive-definite symmetric matrices.

> How would you rephrase the documentation for `eigh` similar to this?

s/Cholesky decomposition/eigenvalue decomposition of symmetric matrices/
s/positive-definite//

I admit that it is superfluous to call the linear subspace Sym(n) of R^{n×n} a Riemannian submanifold of R^{n×n}.

Reply:
Thanks for the suggestion. Could you just elaborate what you mean with `We, however, do not really backpropagate this very process.`? It sounded a bit unnatural to me.

Reply:
A sentence like "LAPACK reconstructs a symmetric matrix from a triangular part of the input" should be inserted.  However the word "reconstruct" there does not mean copying on the memory.

Reply:
Updated the notes, please take a look at them.